no_std support?

What is the state of no_std support? It's my understanding that the macros all get fully expanded at compile time so if the generated code has no dependency on std then _in theory_ you could use a macro generated with this crate in a no_std environment.

Having said that, I have no idea how feasible it actually us given that `TokenStream` and `to_string` and such definitely do need std.

The error I get when trying to use this crate in a no_std environment is the expected:
```
   Compiling proc-macro-hack v0.4.0
error[E0463]: can't find crate for `std`
  |
  = note: the `thumbv6m-none-eabi` target may not be installed
```

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -156,6 +156,8 @@
 //! RESULT
 //! ```
 
+#![no_std]
+
 // Allow the "unused" #[macro_use] because there is a different un-ignorable
 // warning otherwise:
 //

